packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -16,29 +26,35 @@ interface SoqlEditorEvent {
 export enum MessageType {
   ACTIVATED = 'activated',
   QUERY = 'query',
+  SOBJECT_METADATA_REQUEST = 'sobject_metadata_request',
+  SOBJECT_METADATA_RESPONSE = 'sobject_metadata_response',
+  SOBJECTS_REQUEST = 'sobjects_request',
+  SOBJECTS_RESPONSE = 'sobjects_response',
```

Comment:
Protocol for UI to request list of SObject names or particular SObject metadata from VSCode extension.